cmd/state-svc/internal/rtwatcher/watcher.go

```diff
@@ -97,7 +99,11 @@ func (w *Watcher) check() {
 
 func (w *Watcher) RecordUsage(e entry) {
 	logging.Debug("Recording usage of %s (%d)", e.Exec, e.PID)
-	w.an.EventWithSource(anaConst.CatRuntimeUsage, anaConst.ActRuntimeHeartbeat, anaConst.SrcExecutor, e.Dims)
+	source := anaConst.SrcExecutor
+	if filepath.Base(e.Exec) == constants.StateCmd+osutils.ExeExt {
+		source = anaConst.SrcStateTool
+	}
+	w.an.EventWithSource(anaConst.CatRuntimeUsage, anaConst.ActRuntimeHeartbeat, source, e.Dims)
```

Comment:
This feels too error prone to me. Could we instead add the source as an argument for the ReportRuntimeUsage resolver on the svc?

The only place executors call this is here:

https://github.com/ActiveState/cli/blob/be9579908d5177b979384030e17906609b3645df/internal/svcctl/comm.go#L142-L142

Any other place would be state tool. I'd be fine defaulting to state tool if the source is empty.